numpy/_core/include/numpy/ndarraytypes.h

```diff
@@ -578,6 +578,11 @@ typedef struct {
 #define PyDataType_REFCHK(dtype) \
         PyDataType_FLAGCHK(dtype, NPY_ITEM_REFCOUNT)
 
+
+#if !defined(NPY_INTERNAL_BUILD) && NPY_INTERNAL_BUILD
```

Comment:
Shouldn't this be just
```suggestion
#if !defined(NPY_INTERNAL_BUILD)
```

Reply:
Thanks, this might be the issue about `random` requiring changes in the `init.pyd`!  

Unless you disagree, will just add the brackets around it, since checking both seems the pattern we use everywhere (I guess it forces setting it to an actual value which seems a good habit).

Reply:
ok

Reply:
Cool, that was the reason for the Cython/random oddity.  I added new commits for the changes for review, but happy to squash-merge.